lib/ZwaveClient.js

```diff
@@ -1571,22 +1571,27 @@ ZwaveClient.prototype.getGroups = async function (
   const zwaveNode = this.getNode(nodeId)
   const node = this.nodes.get(nodeId)
   if (node && zwaveNode) {
-    let groups = []
+    let endpointGroups = []
     try {
-      groups = await this.driver.controller.getAssociationGroups(nodeId)
+      endpointGroups = await this.driver.controller.getAllAssociationGroups(
+        nodeId
+      )
     } catch (error) {
       logger.warn(`Node ${nodeId} doesn't support groups associations`)
       // node doesn't support groups associations
     }
-    for (const [groupIndex, group] of groups) {
-      // https://zwave-js.github.io/node-zwave-js/#/api/controller?id=associationgroup-interface
-      node.groups.push({
-        text: group.label,
-        value: groupIndex,
-        maxNodes: group.maxNodes,
-        isLifeline: group.isLifeline,
-        multiChannel: group.multiChannel
-      })
+    for (const [endpoint, groups] of endpointGroups) {
+      for (const [groupIndex, group] of groups) {
+        // https://zwave-js.github.io/node-zwave-js/#/api/controller?id=associationgroup-interface
+        node.groups.push({
```

Comment:
@AlCalzone I think this could have some dups when there are multiple endpoints, right?